### ISSUE TITLE
Drop down menu options select the wrong item or do not respond to mouse clicks

### ISSUE THREAD

**Issue by chambers** [aurelienpierreeng/ansel#206]
## Description of the bug

When an item in a drop down menu is selected with the mouse, the application selects the wrong menu item OR the application does not select any item in the list and leaves the drop down menu open. Using the scroll wheel works as a workaround to this issue.

**To Reproduce**

The bug is prevalent throughout the entire application. Select any drop down menu within a module, such as the Input Color Profile module. Try to change the working profile to something else, such as Linear ProPhoto RGB. The application will occasionally select the wrong menu item, or it will not select any item at all.

## Expected behavior

Expected to use the left mouse button to select a module drop down menu item.

## Context

Standard use of the application. The bug makes the application very difficult and cumbersome to use.

## System

I used AUR to install, package: ansel-git

* Ansel version : 0.0.0+388~g43e4fcebe
* OS                :  Linux 6.4.12-arch1-1 x86_64
* Linux - Distro    :  EndeavourOS
* Memory            : 8GB
* Graphics card     : AMD Radeon 7340 (old laptop)
* Graphics driver   : radeon
* OpenCL installed  : N/A
* OpenCL activated  : N/A
* Desktop           : XFCE 4.18
* GTK+              : v3.24.38
* gcc               : 13.2.1 20230801

**Additional context**

 - Can you reproduce with another darktable version(s)? **no**
 - Can you reproduce with a RAW or Jpeg or both? N/A
 - Are the steps above reproducible with a fresh edit (i.e. after discarding history)? N/A
 - If the issue is with the output image, attach an XMP file if (you'll have to change the extension to `.txt`)
 - Is the issue still present using an empty/new config-dir (e.g. start darktable with --configdir "/tmp")? Yes, I only just installed the application.
 - Do you use lua scripts? No

**Comment by Jiyone** [comment]
It looks like a bad division on the recorded x position of the cursor

**Comment by dnlyko** [comment]
Just wondering if this is going to be fixed.  On the windows version it is almost impossible to select 
Guided Laplacians in 'Highlight Reconstruction' or
Aspect in the 'Crop' module.

**Comment by Jiyone** [comment]
@dnlyko you can use the arrow keys and enter to select an option in the comboboxes ;)

**Comment by pjhyor7** [comment]
I have the same problem on Windows 10 on Ansel v 06d9cda. As said this is a very interfering problem. It can be overcome by scrolling the mouse wheel once option is selected, but this isn't ideal and not as able to select the exact option you want by just clicking.

**Comment by vtyrtov** [comment]
The cause of the problem is the `INNER_PADDING` constant in the `src/bauhaus/bauhaus.c `file. Setting the value to 0 solves it.

**Comment by dnlyko** [comment]
> The cause of the problem is the `INNER_PADDING` constant in the `src/bauhaus/bauhaus.c `file. Setting the value to 0 solves it.

Is this issue going to be addressed at development level or does the end-user need to make a change locally?

**Comment by Jiyone** [comment]
> The cause of the problem is the `INNER_PADDING` constant in the `src/bauhaus/bauhaus.c `file. Setting the value to 0 solves it.

This is absolutely not the cause of the issue, this has not been changed since 4 years.
There are some lines using `INNER_PADDING` that have been modified by @aurelienpierre and some other not, so your fix might introduce issues.

**Comment by vtyrtov** [comment]
> > The cause of the problem is the `INNER_PADDING` constant in the `src/bauhaus/bauhaus.c `file. Setting the value to 0 solves it.
> 
> This is absolutely not the cause of the issue, this has not been changed since 4 years. There are some lines using `INNER_PADDING` that have been modified by @aurelienpierre and some other not, so your fix might introduce issues.

@Jiyone, I agree, but at least it made it possible to use the filter menu. I hope in the future the problem will be solved by the author.

**Comment by Jiyone** [comment]
A solution could be to remove `INNER_PADDING` from line 2028.

**Comment by dnlyko** [comment]
> A solution could be to remove `+ INNER_PADDING` from line 2028.

From what file in what folder (tree please) ?

**Comment by Jiyone** [comment]
From src/bauhaus/bauhaus.c

**Comment by dnlyko** [comment]
> From src/bauhaus/bauhaus.c

On my Windows 11 PC there are no folders or files that are named 'src/bauhaus/bauhaus.c'

**Comment by Jiyone** [comment]
@dnlyko yes this is for people who compile. 🤓

**Comment by AlynxZhou** [comment]
I'll have a try and look at the code if I have time.

**Comment by lukadh** [comment]
Hey @AlynxZhou,
from what I have gathered in the matrix chat @aurelienpierre is already knees deep in fixing this issue as part of a bigger rewrite. So you might want to check with him before investing a lot of time

**Comment by AlynxZhou** [comment]
@lukadh Thanks!

**Comment by AlynxZhou** [comment]
> A solution could be to remove `+ INNER_PADDING` from line 2028.

I confirm this is a workaround.

**Comment by aurelienpierre** [comment]
Fixed by 48a86a9c0353f87444fd6956f087af933ecad599

**Comment by AlynxZhou** [comment]
Just build <https://github.com/aurelienpierreeng/ansel/commit/57ed58dd565a6c23224cf9a6d2dc63e096118a36> and I confirm this has been fixed! Good job!

Also I really like the taller line height of menu item now because it is easier to click.

**Comment by dnlyko** [comment]
Does not work properly in Crop module.  Stuck on Freehand

**Comment by lukadh** [comment]
Same here on Pop_OS self-built 57ed58d 

In the crop module I can only change the ratio by mousewheel or arrows, but not selecting a list item by clicking. I have only encountered the issue in the crop module so far.

**Comment by lukadh** [comment]
The selection seems to be fixed with either dbf4f4f89e05e80c7611519f530cd5f653d3c999 or 633d9b71a9e52eb205e26f0f7c6e837b3a3b89ea
